lib/Sema/CSSyntacticElement.cpp

```diff
@@ -1063,6 +1063,15 @@ class SyntacticElementConstraintGenerator
       return;
     }
 
+    // If we're doing code completion, we can skip type-checking the branches
+    // of a SingleValueStmtExpr that don't involve a single expression, and
+    // don't have a code completion token, as they don't contribute to the type.
+    if (cs.isForCodeCompletion() && !braceStmt->getSingleActiveExpression() &&
+        locator->isForSingleValueStmtConjunctionOrBrace() &&
+        !cs.containsIDEInspectionTarget(braceStmt)) {
+      return;
```

Comment:
Could this filtering be done earlier while generate constraints for if/switch expression?

Reply:
How does this look?

Reply:
Yeah that's what I was thinking

Reply:
Okay final version 🤞

Reply:
Looks great, thank you!